lib/Installation/ModuleRegistration/SeparateRegCodesController.pm

```diff
@@ -46,22 +46,11 @@ sub wait_registration_common_regcode_finished {
     }, %$args);
 }
 
-sub add_separate_we_registration_code {
-    my ($self, $regcode) = @_;
-    $self->get_module_regcode_page()->set_we_regcode($regcode);
-    $self->get_module_regcode_page()->press_next();
-}
-
-sub add_separate_ha_registration_code {
-    my ($self, $regcode) = @_;
-    $self->get_module_regcode_page()->set_ha_regcode($regcode);
-    $self->get_module_regcode_page()->press_next();
-}
-
-sub add_separate_ltss_registration_code {
-    my ($self, $regcode) = @_;
-    $self->get_module_regcode_page()->set_ltss_regcode($regcode);
+sub add_separate_registration_code {
+    my ($self, $addon, $regcode) = @_;
+    $self->get_module_regcode_page()->set_regcode($addon, $regcode);
     $self->get_module_regcode_page()->press_next();
+    $self->trust_gnupg_key() if ($addon eq 'we');
```

Comment:
thinking twice, doing this here is weird, it is part of the test, conditional part but not something hidden, we have done wrong this in the past, you will see many example of that in expert partitioner. Let's do it at the level of test (sorry for confusing you here).

Reply:
this is not resolved, but we can do later.